middleware/gateway/unittest/source/test_manager.cpp

```diff
@@ -1271,6 +1271,86 @@ namespace casual
 
       }
 
+      TEST( gateway_manager_outbound, transaction_resource_error_reply_on_remote_domain_shutdown_before_commit_request)
+      {
+         common::unittest::Trace trace;
+
+         auto b = local::domain( R"(
+domain: 
+   name: B
+   groups:
+      -  name: user
+         dependencies: [ base]
+   
+   servers:
+      - path: ${CASUAL_MAKE_SOURCE_ROOT}/middleware/example/server/bin/casual-example-server
+        memberships: [ user]
+
+   gateway:
+      inbound:
+         groups:
+            -  connections: 
+                  -  address: 127.0.0.1:7012
+)");
+
+         // get a handle for b:s inbound
+         auto inbound = unittest::fetch::until( unittest::fetch::predicate::listeners( 1)).inbound.groups.at( 0).process;
+         ASSERT_TRUE( inbound);
+
+         auto a = local::domain( R"(
+domain: 
+   name: A
+   gateway:
+      outbound:
+         groups:
+            -  connections:
+                  -  address: 127.0.0.1:7012
+                     services: [ casual/example/domain/echo/B]
+)");
+
+         unittest::fetch::until( unittest::fetch::predicate::outbound::connected( 1));
+         
+         // call service in B in transaction
+         {
+            using namespace std::literals;
+
+            EXPECT_EQ( transaction::context().begin(), code::tx::ok);
+
+            buffer::Payload payload;
+            payload.type = "X_OCTET/";
+            common::algorithm::copy( "casual"sv, std::back_inserter( payload.memory));
+
+            auto result = common::service::call::context().sync( "casual/example/domain/echo/B", common::buffer::payload::Send{ payload}, {});
+
+            EXPECT_TRUE( result.buffer.memory == payload.memory);
+
+         }
+
+         // b shuts down
+         {
+            communication::device::blocking::send( b.handle().ipc, common::message::shutdown::Request{});
+            unittest::fetch::until( unittest::fetch::predicate::outbound::connected( 0));
+         }
```

Comment:
It's still a timing issue. `gateway-manager` in B will start `b` as soon as it can. `fetch::until` might miss the _down time_. I'm guessing that it's very unlikely, but I'v been more surprised for less... I think you need to make sure that `b` is not B's inbound any more. You might need to _switch/activate to B_ , check, and then switch back to A.

Reply:
I figured (and observed) that we might miss the down time if we just sent a terminate signal to the inbound since b's dm will just restart it. I figured that stopping the entire b domain would solve it, is that not the case? I might also argue that stopping the entire domain has the further 'advantage' of being more similar to the scenario where the issue was discovered.